plugins/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3Service.java

```diff
@@ -57,7 +57,7 @@
      * Client settings derived from those in {@link #staticClientSettings} by combining them with settings
      * in the {@link RepositoryMetadata}.
      */
-    private volatile Map<S3ClientSettings, Map<RepositoryMetadata, S3ClientSettings>> derivedClientSettings = emptyMap();
+    private volatile Map<S3ClientSettings, Map<Settings, S3ClientSettings>> derivedClientSettings = emptyMap();
```

Comment:
Keying this by metadata was just plain wrong. The whole point of this map was to not duplicate `S3ClientSettings` but no two repos will ever share the exact same metadata obviously ... moving this to settings since only the `Settings` in the metadata go into the derived settings anyway.